app/graphql/context.tsx

```diff
@@ -1,23 +1,16 @@
 import { ReactNode } from "react";
 import { createClient, Provider } from "urql";
 import { GRAPHQL_ENDPOINT } from "../domain/env";
-import { defaultLocale } from "../src";
+import { useLocale } from "../src";
 
-const client = createClient({
-  url: GRAPHQL_ENDPOINT,
-  fetchOptions: () => {
-    const lang =
-      typeof document !== "undefined"
-        ? document.querySelector("html")?.getAttribute("lang")
-        : undefined;
-    return {
-      headers: {
-        "Accept-Language": lang ? lang : defaultLocale,
-      },
-    };
-  },
-});
+const client = createClient({ url: GRAPHQL_ENDPOINT });
 
 export const GraphqlProvider = ({ children }: { children: ReactNode }) => {
+  const locale = useLocale();
+
+  client.fetchOptions = () => ({
+    headers: { "Accept-Language": locale },
+  });
```

Comment:
I think this should be wrapped into a useEffect with locale as depedency, for clarity (we are effecting something).

Reply:
Cannot use useEffect since the setting of fetchOptions will be too late (useEffects are run asynchronously), and maybe network calls (with wrong locale) will already have been fired by children.

Reply:
Nit: Maybe useMemo could be used here to have the clarity that fetchoptions only changes when locale changes.

Reply:
I added useCallback there, thanks for the tip!